### PR TITLE
can adjust camera move speed by holding right mouse button and scroll

### PR DESCRIPTION
**Pull request by boooooommmmmm**

### CODE REVIEW

**Comment by Ol6rin** [comment]
Thank you very much for contributing your code!
There're only a few coding style concerns with the PR.
After resolved the concerns, the code can be merged.

File: engine/source/editor/source/editor_ui.cpp [hunk 1]
```diff
@@ -939,10 +945,18 @@ namespace Pilot
...
-            m_tmp_uistate->m_editor_camera->zoom((float)yoffset * 2.0f);
+            if (m_io->isMouseButtonDown(GLFW_MOUSE_BUTTON_RIGHT))
+            {
+                if (yoffset > 0)
+                    m_camera_speed *= 1.2;
+                else
+                    m_camera_speed *= 0.8;
+            }
+            else
+                m_tmp_uistate->m_editor_camera->zoom((float)yoffset * 2.0f);// wheel scrolled up = zoom in by 2 extra degrees
```
[the remark this answers is not in the document]
Comment (Ol6rin) [reply engine/source/editor/source/editor_ui.cpp:957]: Oh, I'm sorry that clang-format file needs some tuning.
Please replace this part as:
```
if (m_io->isMouseButtonDown(GLFW_MOUSE_BUTTON_RIGHT))
{
    if (yoffset > 0)
    {
        m_camera_speed *= 1.2f;
    }
    else
    {
        m_camera_speed *= 0.8f;
    }
}
else
{
    m_tmp_uistate->m_editor_camera->zoom((float)yoffset * 2.0f);// wheel scrolled up = zoom in by 2 extra degrees
}
```

[the remark this answers is not in the document]
Comment (boooooommmmmm) [reply engine/source/editor/source/editor_ui.cpp]: Thx so much, I am going to fix it now. :P

[the remark this answers is not in the document]
Comment (Ol6rin) [reply engine/source/editor/source/editor_ui.cpp]: I found out that I can commit directly, so I have done it. Thank you very much!

**Review by Ol6rin — APPROVED** [review]

**Comment by boooooommmmmm** [comment]
Thx for your correction. :P

**Comment by Ol6rin** [comment]
You're welcome. Thank you for your contribution to Pilot!♥

**Pull request merged**